lib/jsonwp-proxy/proxy.js

```diff
@@ -144,46 +150,55 @@ class JWProxy {
     }
 
     log.debug(`Proxying [${method} ${url || '/'}] to [${method} ${newUrl}] ` +
-             (body ? `with body: ${_.truncate(JSON.stringify(body), {length: LOG_OBJ_LENGTH})}` : 'with no body'));
+      (body ? `with body: ${_.truncate(_.isString(body) ? body : JSON.stringify(body),
+      {length: LOG_OBJ_LENGTH})}` : 'with no body'));
 
-    let res, resBody;
+    const throwProxyError = (error) => {
+      const message = `The request to ${url} has failed`;
+      const err = new Error(message);
+      err.message = message;
+      err.error = error;
+      err.statusCode = 500;
+      throw err;
+    };
     try {
-      res = await this.request(reqOpts);
-      resBody = res.body;
-      log.debug(`Got response with status ${res.statusCode}: ${_.truncate(JSON.stringify(resBody), {length: LOG_OBJ_LENGTH})}`);
-      if (/\/session$/.test(url) && method === 'POST') {
-        if (res.statusCode === 200) {
-          this.sessionId = resBody.sessionId;
-        } else if (res.statusCode === 303) {
```

Comment:
you are intentionally removing support for the older redirect style of session start?

Reply:
I actually wanted to know if it is still used anywhere. As far as I can see this is now obsolete

Reply:
are we sure no supported chromedrivers do this? i remember we had to add it for one of the drivers we proxy to, i'm not sure which one.

Reply:
I would say it would be smarter to make `request` itself to follow redirects rather than adding a special if case. Redirects should be transparent for the proxy client

Reply:
makes sense to me.

Reply:
> I would say it would be smarter to make request itself to follow redirects rather than adding a special if case. Redirects should be transparent for the proxy client

Has this been done, then?

Reply:
no, I didn't spend time on this. I assume this code is obsolete now and is not used anywhere. In general request promise should throw an error if POST request request return 300+ code: https://github.com/request/request-promise/issues/34

I'm going to take care about possible consequences in beta if there are any, but for now I don't expect such

Reply:
Ok. I guess if this gets in and we watch for odd behaviours, I'm ok.